frontend-angularjs/app/scripts/controllers/schedulesCtrl.js

```diff
@@ -1,3 +1,219 @@
 'use strict';
 
-angular.module('hermesApp').controller('SchedulesCtrl', function () {});
+angular.module('hermesApp').controller('SchedulesCtrl', function ($scope, $http, $uibModal, $window, env) {
+
+  var selectedDays = [];
+  function fetchSchedules(pageIndex, callback, pageSize) {
+    pageSize = pageSize || DEFAULT_PAGE_SIZE;
+    $http.get(env.backendBaseUrl + "/schedules?size=" + pageSize + "&page=" + pageIndex)
+      .then(function (response) {
+        callback(response.data);
+      });
+  }
+
+  function fetchLines(callback) {
+    $http.get(env.backendBaseUrl + "/lines").then(function(response) {
+      callback(response.data._embedded.lines);
+    });
+  };
+
+  function formatTime(time) {
+    return (time.getHours()<10?"0":"") + time.getHours() + ":" + (time.getMinutes()<10?"0":"") + time.getMinutes();
+  }
+
+  function persistSchedule (line, desc, startDate, endDate, startTimeInbound, startTimeOutbound, endTime, headway,
+                            dwellTime, averageSpeed, minLayover, callback) {
+    const reqData = {
+      availability: {
+        weekDays: selectedDays,
+        startDate: startDate,
+        endDate: endDate,
+        exceptionDays: []
+      },
+      line: line,
+      description: desc,
+      startTimeInbound: formatTime(startTimeInbound),
+      startTimeOutbound: formatTime(startTimeOutbound),
+      endTimeInbound: formatTime(endTime),
+      endTimeOutbound: formatTime(endTime),
+      headway: headway * 60,
+      dwellTime: dwellTime,
+      averageSpeed: averageSpeed + ' km/h',
+      minLayover: minLayover * 60
+    };
+
+    $http.post(env.backendBaseUrl + "/schedules", reqData)
+      .then(function (response) {
+        if (typeof callback === 'function') {
+          callback({href: response.headers("Location")});
+        }
+      }, function (error) {
+        if (typeof callback === 'function') {
+          callback({error: error});
+        }
+      });
+  };
+
+  function addModalAlert(message, type) {
+    type = type || "danger";
+    $scope.modalAlerts.push({message:message, type:type});
+    $("#schedule").scrollTop(0);
+  };
+
+  function validateSchedule(schedule) {
```

Comment:
use angular directives or html5 to validate form.
https://www.infragistics.com/community/blogs/dhananjay_kumar/archive/2015/08/26/validating-user-input-on-a-form-in-angular-js.aspx